Cleaned up renderers cache and added unit tests

imalberto opened the pull request.

imalberto committed:
FIX for caching renderers for affinity server only

imalberto committed:
FIX for caching renderers for affinity server only

imalberto committed:
added unit tests for caching renderers

imalberto committed:
added unit tests for caching renderers

drewfish commented on lib/app/addons/ac/output-adapter.common.js:
```diff
@@ -201,23 +200,24 @@ YUI.add('mojito-output-adapter-addon', function(Y, NAME) {
+            if (typeof context.runtime !== undefined && 'server' === context.runtime) {
+                renderer = CACHE.renderers[mojitView.engine];
+                if (!renderer) {
+                    CACHE.renderers[mojitView.engine] = renderer =
```
Does jslint complain about this double assignment?

drewfish commented on lib/app/addons/ac/output-adapter.common.js:
```diff
@@ -201,23 +200,24 @@ YUI.add('mojito-output-adapter-addon', function(Y, NAME) {
+                renderer = CACHE.renderers[mojitView.engine];
+                if (!renderer) {
+                    CACHE.renderers[mojitView.engine] = renderer =
+                        new (Y.mojito.addons.viewEngines[mojitView.engine])('', {});
```
It looks like, on the server, that Y.mojito.ViewRenderer isn't being used. That's slightly inconsistent, but it looks like it'll be OK.

drewfish commented:
It looks like the view engines are no longer getting `viewEngineOptions` when run on the server.

drewfish commented on lib/app/autoload/dispatch.common.js:
```diff
@@ -5,7 +5,7 @@
 
 /*jslint anon:true, sloppy:true, nomen:true*/
-/*global YUI*/
+/*global YUI,window*/
```
This seems kinda random.  What does app/autoload/dispatch.common.js have to do with view engines?

imalberto committed:
- nice catch about the viewEngineOptions
- jslint did not complain about the double assignment
- undo the change to dispatch.common. nothing to do with output adapter.

imalberto committed:
- nice catch about the viewEngineOptions
- jslint did not complain about the double assignment
- undo the change to dispatch.common. nothing to do with output adapter.

imalberto commented:
@drewfish
- nice catch about the viewEngineOptions
- mojito jslint did not complain about the double assignment
- undo the change to dispatch.common. nothing to do with output adapter.

drewfish commented:
Could you perhaps add a test, to make sure that `viewEngineOptions` doesn't get lost again in the future?

imalberto committed:
- test case for passing viewEngine options
- change how appConfig is accessed

imalberto committed:
- test case for passing viewEngine options
- change how appConfig is accessed

drewfish committed:
Merge pull request #538 from drewfish/perf-unit-middleware

fixed middleware unit tests

drewfish commented:
+1

imalberto committed:
merging with origin/fix-renderercache
